### DISCUSSION TITLE
Will the baseline code of WikiKG90Mv2 be released?

### DISCUSSION THREAD

**Discussion by chocolate9624** [snap-stanford/ogb#280] [WikiKG90M-LSC]
Hi, I noticed that the baseline code is unavailable. Will it be released?

Thanks.

**Comment by hyren** [comment]
Hi, thanks for your interest. Yes we are cleaning up the code, and will release it by the end of this week.

**Reply by chocolate9624** [reply]
Thanks for your quick response!

**Comment by hyren** [comment]
Please find the implementation of the baselines and instructions [here](https://github.com/google-research/smore/blob/wikikgv2/README_wikikgv2.md) in the wikikgv2 branch of SMORE. Let me know if you have any questions.

**Reply by GeneralBender** [reply]
Hi,

thanks a lot for providing the code! I got a question regarding the evauation of the wikikgv2. In the dataloader you mention:
"Since no candidates are provided by the original dataset, we generate candidates based on heuristics such as degrees / entity types."
I am wondering what exactly these heuristics are and if you sample 1001 candidates as in the first version of the WikiKG90M.
This might be important for the comparison of results.

Any information is appreciated!

Kind regards!

**Reply by hyren** [reply]
Hi, we used the heuristics in the winner solutions of the OGB-LSC KDD cup. For a given (h, r) pair, we pick tail candidates t by choosing the entities with at least one edge (_, r, t) on the graph, and sort all the tails by its degree. We select the 20k tails with highest degree.
